### DISCUSSION TITLE
API support for the ComfyUI part

### DISCUSSION THREAD

**Discussion by JuMayr** [Stability-AI/StableSwarmUI#298] [General]
Hi, i love your project and your documentation :) 
One question: Does StableSwarmUI also support the distribution of ComfyUI API calls over multiple GPUs?

**Comment by mcmonkey4eva** [comment]
Ideally use the Swarm API directly, at which point yes, splitting your jobs between all available backends will automatically happen.
You can feed custom comfy workflows through the regular Swarm API if you want -- just run it in-browser, load things up how you want (eg make a workflow in comfy tab and click 'export to generate tab') and do a generation on the Generate tab, and open your browser console (F12) and view the network traffic. The default UI generation request goes via a WebSocket connection to `/API/GenerateText2ImageWS` but you can send the exact same data in the exact same json format over HTTP to `/API/GenerateText2Image` (ie just remove the "WS" on end) instead.

Example external program hitting Swarm API: https://github.com/mcmonkey4eva/SimpleDiscordAIBot/blob/8c53541a8f589e38a7d68bf027175424bc5ef66f/Program.cs#L82-L163

If you're stuck just using ComfyUI API calls directly, you can hit `/ComfyBackendDirect`, and there is support for multi-GPU (if you hit `/ComfyBackendDirect/prompt` with a cookie in headers of `comfy_domulti=true` it will redirect dynamically to whichever backend has the fewest jobs in queue, though that requires you have a `client_id` corresponding to a websocket connection) (This setup was built for the comfy UI layer to do it, but you can easily enough do it as an api call)
